### DISCUSSION TITLE
(Auto) Refresh for multiple panels in Insights Module

### DISCUSSION THREAD

**Discussion by DanielBiegler** [directus/directus#11824] [Feature Requests]
Hi there!

I set up some Dashboards with the Insights Module and wondered why there's no "Auto Refresh" Option on the right Sidebar like in the Content-Module?

I think this feature could be neat and makes sense, since the content module is also able to refresh.

I stumbled upon this idea because I have two panels which display similar data like so:

Two metric panels, one list panel. One is total count of entries in a collection, the other one are entries of the same collection but filtered and the list holds said filtered entries to be able to open them directly.

![image](https://user-images.githubusercontent.com/14810858/155378805-2fea0ff0-8b24-45ec-8ca7-725862f4760b.png)

This works, but when you update entries via the list panel, the two metric panels do not update. See:

![image](https://user-images.githubusercontent.com/14810858/155379418-9722327d-1e0c-4cd7-ba18-b36fe556068b.png)

I searched for different terms in the issues, discussions and PRs but couldnt find out more. @rijkvanzanten and @azrikahar worked on #10180 which is similar to what I'm about here.

So the question is really whether you would like to have

- completely reactive dashboards where actions may update multiple panels
- an "auto-refresh" feature just like in the Content-Module
- a manual refresh button besides the edit button, see the following professional illustration

![image](https://user-images.githubusercontent.com/14810858/155381874-d3ede120-40de-4837-b526-cfcf94e4b8fa.png)

I haven't contributed to Directus (yet) and this might be something to get me started if you're interested. Since I'm new I might need a little help though.

**Comment by rijkvanzanten** [comment]
> I set up some Dashboards with the Insights Module and wondered why there's no "Auto Refresh" Option on the right Sidebar like in the Content-Module?
I think this feature could be neat and makes sense, since the content module is also able to refresh.

I fully agree! The only reason that doesn't exist (yet) is that the loading of the page works slightly differently. The layouts are a single data set that's being downloaded, which can then be refreshed by asking the layout to refresh the data. In insights however, each panel has a custom data source, which in turn means that it's technically a little more complicated to tell each of the panels to update the data. We might want to think through adding some sort of centralized data downloading pipeline that can then refresh the data on behalf of the panels, which then allows us to more easily add a live-reload (and later real-time) updating in that pipeline 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
